homeassistant/components/sky_remote/config_flow.py

```diff
@@ -0,0 +1,59 @@
+"""Config flow for sky_remote."""
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.helpers.selector import selector
+
+from .const import DOMAIN, SKY_REMOTE_CONFIG_SCHEMA
+
+
+class SkyRemoteConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Sky Remote."""
+
+    VERSION = 1
+    MINOR_VERSION = 1
+    DATA_SCHEMA: dict[Any, Any] = {
+        **SKY_REMOTE_CONFIG_SCHEMA,
+        "legacy_port": selector({"boolean": None}),
+    }
```

Comment:
Please set it outside of the config flow. What is `legacy_port`?

Reply:
Do you mean moving outside of the class or into another file (const.py?)
Legacy port is used when the device is running old firmware which means the port will be different to default. 
It is explained in strings.json, comments about it's purpose in code required?

Reply:
Is it possible to have it automatically checking the correct port?

Reply:
Changed to auto detect port to use.